components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/user/OpenIDConnectUserEndpoint.java

```diff
@@ -70,6 +71,12 @@ public Response getUserClaims(@Context HttpServletRequest request) throws OAuthS
                     UserInfoEndpointConfig.getInstance().getUserInfoAccessTokenValidator();
             OAuth2TokenValidationResponseDTO tokenResponse = tokenValidator.validateToken(accessToken);
 
+            if (tokenResponse.isValid()) {
+                if (!isValidTokenBinding(tokenResponse.getTokenBinding(), request)) {
```

Comment:
shall we use one if condition for this with a AND operation?

Reply:
Done